language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/util/SnippetGenerator.java

```diff
@@ -1480,8 +1497,9 @@ public static SnippetBlock getOnFailClauseSnippet() {
         String snippet = "on fail ${1:var} ${2:varName} {" + CommonUtil.LINE_SEPARATOR + "\t${3}"
                 + CommonUtil.LINE_SEPARATOR + "}";
 
-        return new SnippetBlock(ItemResolverConstants.ON_FAIL_CLAUSE, snippet, ItemResolverConstants.SNIPPET_TYPE,
-                Kind.SNIPPET);
+        return new SnippetBlock(ItemResolverConstants.ON_FAIL_CLAUSE, generateFilterText(Arrays.asList(
+                ItemResolverConstants.ON_FAIL_CLAUSE, ItemResolverConstants.FAIL_KEYWORD)),
```

Comment:
What if we add `on` and `fail` keywords separately instead of on fail clause?

Reply:
Done